app/com/linkedin/drelephant/spark/heuristics/ExecutorsHeuristic.scala

```diff
@@ -163,9 +166,9 @@ object ExecutorsHeuristic {
       severityOfDistribution(storageMemoryUsedDistribution, ignoreMaxBytesLessThanThreshold)
 
     lazy val taskTimeDistribution: Distribution =
-      Distribution(executorSummaries.map { _.totalDuration })
+      Distribution(executorSummaries.map {executionSummary => if(executionSummary.endTime == null) (appAttemptInfo.endTime.getTime - executionSummary.addTime.getTime) else (executionSummary.endTime.getTime - executionSummary.addTime.getTime) })
```

Comment:
Avoid using null, if-else can be converted to getOrElse, and avoid very long lines (break into multiple lines so that it can be read more easily without having to scroll).